File: app/src/main/java/com/faltenreich/diaguard/feature/preference/decimal/DecimalPlacesPreference.java
package com.faltenreich.diaguard.feature.preference.decimal;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.DialogPreference;

import com.faltenreich.diaguard.R;

public class DecimalPlacesPreference extends DialogPreference {

    public DecimalPlacesPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public int getDialogLayoutResource() {
        return R.layout.preference_decimal_places;
    }
}

File: data/src/commonMain/kotlin/com/faltenreich/diaguard/data/preference/decimal/DecimalPlacesPreference.kt
package com.faltenreich.diaguard.data.preference.decimal

import com.faltenreich.diaguard.data.preference.Preference

data object DecimalPlacesPreference : Preference<Int, Int> {

    override val key = "Decimal places"

    override val default = 1

    override val onRead = { decimalPlaces: Int -> decimalPlaces }

    override val onWrite = { decimalPlaces: Int -> decimalPlaces }
}